packages/nextjs/hooks/scaffold-eth/useTransactor.tsx

```diff
@@ -47,21 +46,17 @@ export const useTransactor = (_walletClient?: WalletClient): TransactionFunc =>
     }
 
     let notificationId = null;
-    let transactionHash: Awaited<WriteContractResult>["hash"] | undefined = undefined;
+    let transactionHash: Hash | undefined = undefined;
     try {
       const network = await walletClient.getChainId();
       // Get full transaction from public client
-      const publicClient = getPublicClient();
+      const publicClient = getPublicClient(wagmiConfig);
 
       notificationId = notification.loading(<TxnNotification message="Awaiting for user confirmation" />);
       if (typeof tx === "function") {
         // Tx is already prepared by the caller
         const result = await tx();
-        if (typeof result === "string") {
-          transactionHash = result;
-        } else {
-          transactionHash = result.hash;
-        }
+        transactionHash = result;
```

Comment:
don't we need the **if** anymore?

It was introduced here https://github.com/scaffold-eth/scaffold-eth-2/pull/623

Reply:
> don't we need the if anymore?

Yup 🙌, Wagmi & Viem have now standardized it to always be directly give hash, instead of before in V1 on someplace it was giving direct Hash and sometimes objects with `{hash : ${0xstring} }`

Wagmi Docs: https://wagmi.sh/core/guides/migrate-from-v1-to-v2#updated-sendtransaction-and-writecontract-return-type

Here is working example with v2 and #623 : https://github.com/scaffold-eth/scaffold-eth-2/blob/a9f0d5766c823fcc580c1efdb83f020a19a09bc1/packages/nextjs/app/page.tsx#L113